lib/daru/maths/statistics/vector.rb

```diff
@@ -855,6 +835,14 @@ def raw_sample_without_replacement sample
 
           out.collect { |i| valid[i] }
         end
+
+        def sort_without_nil(&block)
+          dv = reject_values(nil)
```

Comment:
But reflecting the effect of `@data.compact.to_a` correspondingly into the `@index.to_a` is something which could make `index_of_min` and `index_of_max` more ineffecient / unidiomatic. Rather, `reject_values(nil)` seemed like a better alternative. Let me know if it should be reverted back to `@data.data.to_a.compact` approach.